Analysis/interface/TrainTuple.h

```diff
@@ -0,0 +1,262 @@
+/*! Definition of a tuple with all event information that is required for the tau analysis.
+*/
+
+#pragma once
+
+#include "TauMLTools/Core/interface/SmartTree.h"
+#include "TauMLTools/Analysis/interface/TauIdResults.h"
+#include "DataFormats/GeometryVector/interface/GlobalPoint.h"
+#include <Math/VectorUtil.h>
+
+#define TAU_ID(name, pattern, has_raw, wp_list) VAR(std::vector<uint16_t>, name) VAR(std::vector<Float_t>, name##raw)
+#define TAU_VAR(type, name) VAR(std::vector<type>, tau_##name)
+#define CALO_TOWER_VAR(type, name) VAR(std::vector<type>, caloTower_##name)
+#define CALO_TAU_VAR(type, name) VAR(std::vector<type>, caloTau_##name)
+#define HBHE_VAR(type, name) VAR(std::vector<type>, caloRecHit_hbhe_##name)
+#define HO_VAR(type, name) VAR(std::vector<type>, caloRecHit_ho_##name)
+#define HF_VAR(type, name) VAR(std::vector<type>, caloRecHit_hf_##name)
+#define ECAL_VAR(type, name) VAR(std::vector<type>, caloRecHit_ee_##name) VAR(std::vector<type>, caloRecHit_eb_##name)
+#define TRACK_VAR(type, name) VAR(std::vector<type>, track_##name) VAR(std::vector<type>, patatrack_##name)
+#define VERT(type, name) VAR(std::vector<type>, vert_##name) VAR(std::vector<type>, patavert_##name)
+
+
+#define VAR2(type, name1, name2) VAR(type, name1) VAR(type, name2)
+#define VAR3(type, name1, name2, name3) VAR2(type, name1, name2) VAR(type, name3)
+#define VAR4(type, name1, name2, name3, name4) VAR3(type, name1, name2, name3) VAR(type, name4)
+
+#define TAU_DATA() \
+    /* Event Variables */ \
+    VAR(UInt_t, run) /* run number */ \
+    VAR(UInt_t, lumi) /* lumi section */ \
+    VAR(ULong64_t, evt) /* event number */ \
+    VAR(Int_t, module_index) /* number of primary patavertices */ \
+    VAR(Int_t, path_state) /* number of primary patavertices */ \
```

Comment:
- `module_index` -> `defaultDiTauPath_lastModuleIndex ` + fix the comment
- `Int_t, path_state` -> `bool, defaultDiTauPath_result` + fix the comment